lib/puppet/functions/find_template.rb

```diff
@@ -0,0 +1,63 @@
+# Finds an existing template from a module and returns its path.
+#
+# The argument to this function should be a String as a `<MODULE NAME>/<TEMPLATE>`
+# reference, which will search for `<TEMPLATE>` relative to a module's `templates`
+# directory. (For example, the reference `mymod/secret.conf.epp` will search for the
+# file `<MODULES DIRECTORY>/mymod/templates/secret.conf.epp`.)
+#
+# The primary use case is for agent-side template rendering with late-bound variables
+# resolved, e.g. from secret stores inaccessable to the master, such as
+#
+# ````
+# $variables = {
+#   'password' => Deferred('vault_lookup::lookup',
+#                   ['secret/mymod', 'https://vault.example.com:8200']),
+# }
+#
+# # compile the template source into the catalog
+# file { '/etc/secrets.conf':
+#   ensure  => file,
+#   content => Deferred('inline_epp',
+#                [find_template('mymod/secret.conf.epp').file, $variables]),
+# }
+# ```
+#
+#
+#
+# This function can also accept:
+#
+# * An absolute String path, which will check for the existence of a template from anywhere on disk.
```

Comment:
Might be good to clarify _where_ on disk this is searched. The master or the agent?

Reply:
👍